### PR TITLE
Redirect Flow

### PR DESCRIPTION
**Pull request by sw1ld**
# Summary

* Run tests depending on SCA profile
* Adapt existing tests to cover also Redirect Flow

# Checklist for Definition of Done

* [x] Code is technically reviewed by myself
* [x] Code is manually tested
* [x] Pipeline is green
* [x] Automatic tests for new functionality are created
* [x] Documentation is updated
* [x] Branch is rebased and therefore ready to merge
* [x] Git history is clean

### CODE REVIEW

File: service/src/main/java/de/adorsys/psd2/sandbox/xs2a/web/RedirectController.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    this.redirectService = redirectService;
+  }
+
+  /**
+   * Sets status of payment resource to value depending on PSU-ID.
+   *
+   * @param externalId Payment Id
+   * @param psuId      Psu Id
+   */
+  @RequestMapping(value = "/init/pis/{external-id}", params = "psu-id")
```
Comment (atamanroman) [inline service/src/main/java/de/adorsys/psd2/sandbox/xs2a/web/RedirectController.java:27]: Style not consistent with the other params where it's `params = { "psu-id" }` I'd prefer it without braces.

File: service/src/test/java/de/adorsys/psd2/sandbox/xs2a/ais/AisConsentCreationSteps.java [hunk 2]
```diff
@@ -82,28 +85,15 @@ public void psu_created_a_consent_on_dedicated_account(String accounts, String b
...
 
-    context.setConsentId(response.getBody().getConsentId());
-  }
-
-  private List<AccountReference> fillAccountReferences(String[] ibans) {
-    ArrayList<AccountReference> result = new ArrayList<>();
-
-    if (ibans[0].equals("null")) {
-      return null;
+    if (scaApproach.toLowerCase().equals("redirect")) {
```
Comment (atamanroman) [inline service/src/test/java/de/adorsys/psd2/sandbox/xs2a/ais/AisConsentCreationSteps.java]: There is also a `equalsIgnoreCase` method (occurs a few times in this change)

File: service/src/test/java/de/adorsys/psd2/sandbox/xs2a/ais/AisConsentCreationSteps.java [hunk 3]
```diff
@@ -290,35 +298,62 @@ public void psuTriesToAuthoriseConsent(String psuId, String password) {
...
 
     String authorisationId = TestUtils
-        .extractAuthorisationId((String) startScaResponse.getBody().getLinks()
-            .get("startAuthorisationWithPsuAuthentication"));
+        .extractId((String) startScaResponse.getBody().getLinks()
+            .get("startAuthorisationWithPsuAuthentication"), "authorisations");
 
-    PsuData psuData = new PsuData();
-    psuData.setPassword(password);
+    String url = "consents/" + context.getConsentId() + "/authorisations/" + authorisationId;
```
Comment (atamanroman) [inline service/src/test/java/de/adorsys/psd2/sandbox/xs2a/ais/AisConsentCreationSteps.java]: `String.format()` or `UriComponentsBuilder` would be an option. Or `Joiner.on("/").join("consents", context.getConsentId, "authorisations", authorisationId)`.

File: service/src/test/resources/application-test.properties [hunk 4]
```diff
@@ -1,3 +1,3 @@
 spring.datasource.driver-class-name=org.testcontainers.jdbc.ContainerDatabaseDriver
 spring.datasource.url=jdbc:tc:postgresql://localhost/consent
-
+sandbox.scaapproach=redirect
```
Comment (atamanroman) [inline service/src/test/resources/application-test.properties:3]: We should be able to get this value from `ProfileService` without having to duplicate this value. Add at least a TODO here. If we need it in both contexts we have to refactor a bit.

File: service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    aisConsentAuthorizationRepository.save(aisConsentAuth);
+  }
+
+  /**
+   * Sets payment status in CMS depending on PSU-ID.
+   *
+   * @param externalId Payment Id
+   * @param psuId      Psu Id
+   */
+  public void handlePaymentRedirectRequest(String externalId, String psuId, boolean isInit) {
```
Comment (atamanroman) [inline service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java:78]: `isInit` is unused. Why do we need different redirects for cancellation and initiation? How would we handle this in the banking UI?

File: service/src/main/resources/static/thyme.html [hunk 6]
```diff
@@ -0,0 +1,27 @@
+<!--
```
Comment (atamanroman) [inline service/src/main/resources/static/thyme.html]: What's that? AFAIK we do not serve static HTML right now, even if it's in the JAR.

File: service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    PisAuthorization payment = pisAuthorization.get();
+
+    Optional<List<PisPaymentData>> pisPaymentDataList = pisPaymentDataRepository
+        .findByPaymentId(payment.getPaymentData().getPaymentId());
+
+    if (!pisPaymentDataList.isPresent()) {
+      //TODO handle error case
+      return;
+    }
+    PisPaymentData pisPaymentData = pisPaymentDataList.get().get(0);
```
Comment (atamanroman) [inline service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java:96]: Is the list always not empty?

File: service/src/main/java/de/adorsys/psd2/sandbox/xs2a/web/RedirectController.java (hunk 1 above)
Comment (wandi34) [reply service/src/main/java/de/adorsys/psd2/sandbox/xs2a/web/RedirectController.java:27]: done

File: service/src/test/java/de/adorsys/psd2/sandbox/xs2a/ais/AisConsentCreationSteps.java (hunk 2 above)
Comment (wandi34) [reply service/src/test/java/de/adorsys/psd2/sandbox/xs2a/ais/AisConsentCreationSteps.java]: Done

File: service/src/test/java/de/adorsys/psd2/sandbox/xs2a/ais/AisConsentCreationSteps.java (hunk 3 above)
Comment (wandi34) [reply service/src/test/java/de/adorsys/psd2/sandbox/xs2a/ais/AisConsentCreationSteps.java]: Done

File: service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java (hunk 5 above)
Comment (wandi34) [reply service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java:78]: We will need `isInit` to set different status for initiation and cancellation flows.
- Successfull init response results in ACCP
- Successfull cancel response results in CANC

And this is the same procedure in the banking Ui.

I added a TODO for this parameter

File: service/src/main/resources/static/thyme.html (hunk 6 above)
Comment (wandi34) [reply service/src/main/resources/static/thyme.html]: Done

File: service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java (hunk 7 above)
Comment (wandi34) [reply service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java:96]: There is a check for a non existing DataList before this function. So the function should always be called with a filled LIst. 

Please check and resolve the conversation when its okay.

Comment (atamanroman) [reply service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java:96]: We check if there IS a list (not null) but not if its not empty. I've extended the check.

**Pull request merged**